### DISCUSSION TITLE
Disable a specific case provided by cindent when enter a colon `:`

### DISCUSSION THREAD

**Discussion by nyngwang** [neovim/neovim#24566] [Q&A] [answered]
As title. What I have currently after entering `:` after the `std`:

```cpp
  std::memset(tempArr, 0, MAX_LEN_CASE*sizeof(int));
  for (int i=0; i<MAX_LEN_CASE; ++i) {
std:
  }
```

What I expect is:

```cpp
  std::memset(tempArr, 0, MAX_LEN_CASE*sizeof(int));
  for (int i=0; i<MAX_LEN_CASE; ++i) {
    std:
  }
```

If I enter the second `:` after the first one I will get the expected behaviour:

```cpp
  std::memset(tempArr, 0, MAX_LEN_CASE*sizeof(int));
  for (int i=0; i<MAX_LEN_CASE; ++i) {
    std::
  }
```

I'm using Neovim but I found that this is also reproducible in Vim. Can anyone tell me how to disable it or maybe explain what could be the intent of this design?

**Comment by zeertzjq** [comment, marked as answer]
`:h cinkeys-format`

**Reply by nyngwang** [reply]
I tried `vim.opt.cinkeys:remove({ ':' })` and `vim.opt.cinkeys = "0{,0},0),0],0#,!^F,o,O,e"` to remove the `:` but it doesn't work. Could you teach me how to set it correctly?

**Reply by nyngwang** [reply]
Nevermind, I found the answer after trying: `setlocal cinkeys-=:`.
